pos_supplierinfo_search/models/product.py

```diff
@@ -0,0 +1,25 @@
+import json
+
+from odoo import fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    # technical field used in POS frontend
+    supplier_data_json = fields.Char(
+        "Supplier data dict", readonly=True, compute="_compute_supplier_data_json"
+    )
+
+    def _compute_supplier_data_json(self):
+        for t in self:
+            res = []
+            for s in t.seller_ids:
+                res.append(
+                    {
+                        "supplier_name": s.name.display_name,
+                        "supplier_product_code": s.product_code or "",
+                        "supplier_product_name": s.product_name or "",
+                    }
+                )
+            t.supplier_data_json = json.dumps(res)
```

Comment:
```suggestion
        for rec in self:
            rec.supplier_data_json = json.dumps(
                [
                    {
                        "supplier_name": s.name.display_name,
                        "supplier_product_code": s.product_code or "",
                        "supplier_product_name": s.product_name or "",
                    }
                    for s in rec.seller_ids
                ]
            )
```

Not blocking, but you could rewrite it like this for readability